packages/inter-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -1112,6 +1294,26 @@ export const prepareVaultManagerKit = (
               ),
           );
 
+          // helper.makeLiquidationVisibilityWriters and schedulesP depends on others vats,
```

Comment:
> If `schedules` is broken, liquidation might take place once, but we wouldn't be able to schedule the next round. There's little value in attempting to survive in this situation. Similarly, if we can't write to vstorage, we should give up early, since something is misconfigured.

Makes sense 👍 

> The vats are separate address spaces, but not entirely separate responsibilities. In an actually distributed system, it might make sense to survive and carry on if some of the services you rely on can't be reached, but these are not independent services. If they're broken, we're better off stopping and fixing them than trying to carry on.

Also makes sense 👍

Reply:
> I think `writePreAuction()`, `writePostAuction()`, and `writeAuctionResults()`, can be functions directly in the helper facet.

For this one, I believe one reason we chose what we did was to not increment the number of `await`s in `liquidateVaults()` method.  Since we need to create `recorderKits` (thus `storageNode`s) for preAuction, postAuction and auctionResults phases we must do an async call to get the those storage nodes and await them. The reason why we await `E(parentStorageNode).makeChildNode('preAuctionStorageNode...')` calls is that `recorderKit` is a durable exo class and it has `storageNode` in its state. As we cannot store promises durably we have to await those `makeChildNode` calls. See https://github.com/Jorge-Lopes/agoric-sdk/blob/8f17f7076904be8533d748335ddee62573b8de65/packages/inter-protocol/src/vaultFactory/vaultManager.js#L751-L763

So if we are to move away from [makeLiquidationVisibilityWriters()](https://github.com/Jorge-Lopes/agoric-sdk/blob/8f17f7076904be8533d748335ddee62573b8de65/packages/inter-protocol/src/vaultFactory/vaultManager.js#L641) closure, we have add an extra await into `liquidateVaults()` like; 

```js
const liquidationRecorderKits =
            await this.facets.helper.makeLiquidationRecorderKits(timestamp);
```

To make sure we have the storage nodes and recorderKits that we want.

I just wanted to bring attention as to why we didn't just put `writePreAuction()`, `writePostAuction()`, and `writeAuctionResults()` directly to the helper facet. If you say an additional await is not so bad or it's not worth the effort, we can still move the `write...` methods to the helper facet. No issue at all.

Reply:
> one reason we chose what we did was to not increment the number of awaits in liquidateVaults() method. 

That's a helpful goal. Couldn't we get that by storing the recorderKits in `ephemera`? We do have to rebuild the recorderKits in each incarnation, but that doesn't make it necessary to dynamically generate those methods.

Reply:
Okay, good point. That idea won't work, then.

I think the thing that's making my spidey senses tingle is the manual dispatch in `writeLiqVisibility`. Dispatching on a string doesn't seem like the best approach. I see that since liquidationRecorderKits is generated dynamically and might be missing, we have to check before calling it. I'd prefer an explicit check followed by a method call. How about if at 1313 we do this:
```
  if (checkWritersPresent(liquidationVisibilityWriters)) {
    liquidationVisibilityWriters.writePreAuction(vaultData);
  }
```
and at 1356:
```
  if (checkWritersPresent(liquidationVisibilityWriters)) {
    liquidationVisibilityWriters.writeAuctionResults({
       plan,
       totalCollateral,
       totalDebt,
       auctionSchedule,
     });
    liquidationVisibilityWriters.writePostAuction({ plan, vaultsInPlan });
  }
```
I suspect the liquidation mechanism has accumulated enough state that it really ought to be a separate object, but that refactoring should wait for a separate PR.

Reply:
> I think the thing that's making my spidey senses tingle is the manual dispatch in writeLiqVisibility. Dispatching on a string doesn't seem like the best approach.

Yeah, I see what you are saying.  Here are my action items;
* Stop using `writeLiqVisibility` and implement your suggestion
* Refactor [allValuesSetteled](https://github.com/Jorge-Lopes/agoric-sdk/blob/8f17f7076904be8533d748335ddee62573b8de65/packages/inter-protocol/src/vaultFactory/vaultManager.js#L1306) to regular `Promise.all`

Thanks Crhis!